versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1052,7 +1052,8 @@ protected CommitLogEntry buildKeyList(
               if (buildState.embedded) {
                 // filling the embedded key-list in CommitLogEntry
 
-                if (buildState.currentSize + keyTypeSize < config.getMaxKeyListSize()) {
+                if (buildState.currentSize + keyTypeSize
+                    < Math.min(maxEntitySize(), config.getMaxKeyListSize())) {
```

Comment:
nit: extract a method for the `min` logic since it's used in two places?

Reply:
or perhaps fold it into the impl of `maxEntitySize`?

Reply:
latter, also realized that there was another "configurable default" :(